packages/tools/pie-icons-webc/build.js

```diff
@@ -62,6 +61,17 @@ export class ${name} extends LitElement implements IconProps {
             this._svg?.setAttribute('width', svgSize.width);
             this._svg?.setAttribute('height', svgSize.height);
         }
+
+        const hasSvgElement = this.closest('pie-icon-button, pie-button');
+
+        if (hasSvgElement) {
```

Comment:
In terms of doing this through JS – would this work with SSR? (I think Raouf did something similar and said that anything in this JS isn't SSR compatible)

If so, another way of doing this could be to expose some CSS vars for the icon sizes set on the host, so that they can then be overriden from other components.

We do this in the [iconButton SCSS](https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-icon-button/src/iconButton.scss#L3-L10) – give me a shout if you want to chat about it though.

Reply:
So i've been looking at this and before i even get to run it in SSR mode the change seems to not come through at all for some reason. Did Raouf have this issue or was it specifically SSR he has the issue with?

Reply:
Have caught up separately and going with the CSS approach now, so resolving this thread.